uint64 inc
uint8 kemetF
uint8 kemetB